data/helpers.d/filesystem

```diff
@@ -1,57 +1,126 @@
+# backup dir
+# | arg: is_big - 1 to indicate data are big (mail, video, image ...)
+# | arg: arg - Deprecated arg
 ynh_backup() {
-    local SRCPATH=$1
-    local DESTPATH=$2
-    local TO_BIND=${3:-0}
-    local SUDO_CMD="sudo"
-    [[ "${4:-}" = "1" ]] && SUDO_CMD=
+    local SRC_PATH=$1
+    local DEST_PATH=${2:-}
+    local IS_BIG=${3:-0}
 
     # validate arguments
-    [[ -e "${SRCPATH}" ]] || {
-        echo "Source path '${SRCPATH}' does not exist" >&2
+    [[ -e "${SRC_PATH}" ]] || {
+        echo "Source path '${SRC_PATH}' does not exist" >&2
         return 1
     }
 
-    # prepend the backup directory
-    [[ -n "${YNH_APP_BACKUP_DIR:-}" && "${DESTPATH:0:1}" != "/" ]] \
-        && DESTPATH="${YNH_APP_BACKUP_DIR}/${DESTPATH}"
-    [[ ! -e "${DESTPATH}" ]] || {
-        echo "Destination path '${DESTPATH}' already exist" >&2
+    # Transform SRCPATH as absolute path
+    SRC_PATH="$(realpath $SRC_PATH)"
+
+    # If unspecified DESTPATH=SRCPATH
+    [[ -z "$DEST_PATH" ]] && \
+        DESTPATH="${SRC_PATH#/}"
+
+    # Transform DEST_PATH into relative path
+    [[ "${DEST_PATH:0:1}" == "/" ]] \
+        && DEST_PATH="${DEST_PATH#$YNH_CWD/}"
+
+    # Complete DEST_PATH if ended by a /
+    [[ "${DEST_PATH: -1}" == "/" ]] \
+        &&  DEST_PATH="${DEST_PATH}/$(basename $SRC_PATH)"
+
+    # Check if DEST_PATH already exists in tmp archive
+    [[ ! -e "${DEST_PATH}" ]] || {
+        echo "Destination path '${DEST_PATH}' already exist" >&2
         return 1
     }
 
-    # attempt to bind mounting the directory
-    if [[ "${CAN_BIND}" = "1" && "${TO_BIND}" = "1" ]]; then
-        eval $SUDO_CMD mkdir -p "${DESTPATH}"
+    # Write file to backup into backup_list
+    local REL_DIR="${YNH_CWD#$YNH_BACKUP_DIR}"
+    REL_DIR="${REL_DIR%/}/"
+    local SRC=$(echo "${SRC_PATH}" | sed -r 's/"/\"\"/g')
+    local DEST=$(echo "${REL_DIR}${DEST_PATH}" | sed -r 's/"/\"\"/g')
+    DEST="${DEST#/}"
+    echo "\"${SRC}\",\"${DEST}\"" | sudo tee --append ${YNH_BACKUP_CSV} > /dev/null
 
-        if sudo mount --rbind "${SRCPATH}" "${DESTPATH}"; then
-            # try to remount destination directory as read-only
-            sudo mount -o remount,ro,bind "${SRCPATH}" "${DESTPATH}" \
-              || true
-            return 0
-        else
-            CAN_BIND=0
-            echo "Bind mounting seems to be disabled on your system."
-            echo "You have maybe to check your apparmor configuration."
-        fi
+    sudo mkdir -p $(dirname "${DEST_PATH}")
+}
 
-        # delete mountpoint directory safely
-        mountpoint -q "${DESTPATH}" && sudo umount -R "${DESTPATH}"
-        eval $SUDO_CMD rm -rf "${DESTPATH}"
-    fi
+# Restore all files linked to the restore hook or to the restore app script
+#
+# usage: ynh_restore
+#
+ynh_restore () {
+    local REL_DIR="${YNH_CWD#$YNH_BACKUP_DIR/}"
+    REL_DIR="${REL_DIR%/}/"
+    sudo grep -ohP $'^\".*\",\"$REL_DIR.*\"$' ${YNH_BACKUP_CSV} | \
```

Comment:
I guess you want to use `_get_archive_path` also here like in `ynh_restore_file` ?